pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java

```diff
@@ -79,23 +79,36 @@ protected UsernamePasswordCredentials retrieveCredentials(final WebContext conte
             credentials = getCredentialsExtractor().extract(context);
             logger.debug("usernamePasswordCredentials: {}", credentials);
             if (credentials == null) {
-                String redirectionUrl = CommonHelper.addParameter(this.loginUrl, this.usernameParameter, username);
-                redirectionUrl = CommonHelper.addParameter(redirectionUrl, ERROR_PARAMETER, MISSING_FIELD_ERROR);
-                logger.debug("redirectionUrl: {}", redirectionUrl);
-                final String message = "Username and password cannot be blank -> return to the form with error";
-                logger.debug(message);
-                throw RequiresHttpAction.redirect(message, context, redirectionUrl);
+				// it's an AJAX request -> unauthorized (instead of a
```

Comment:
This change addresses a bug I saw when a login to the authentication URL was accessed via an ajax call. When there is an ajax call you can't do a redirect because the browser will redirect before the ajax caller get's control. Hence this fixes that and for ajax it doesn't do a redirect.